services/consensuscontext/validate_block_test.go

```diff
@@ -0,0 +1,130 @@
+package consensuscontext
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/config"
+	"github.com/orbs-network/orbs-network-go/crypto/digest"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+)
+
+// func (s *service) ValidateTransactionsBlock(ctx context.Context, input *services.ValidateTransactionsBlockInput) (*services.ValidateTransactionsBlockOutput, error) {
+func TestValidateTransactionBlock(t *testing.T) {
+
+	log := log.GetLogger().WithOutput(log.NewFormattingOutput(os.Stdout, log.NewHumanReadableFormatter()))
+	metricFactory := metric.NewRegistry()
+	cfg := config.ForConsensusContextTests(nil)
+	transaction := builders.TransferTransaction().WithAmountAndTargetAddress(10, builders.AddressForEd25519SignerForTests(6)).Build()
+	txRootHashForValidBlock, err := CalculateTransactionsRootHash([]*protocol.SignedTransaction{transaction})
+	prevBlock1 := builders.BlockPair().WithHeight(1001).Build()
+	prevBlock2 := builders.BlockPair().WithHeight(1002).Build()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := NewConsensusContext(
+		&services.MockTransactionPool{},
+		&services.MockVirtualMachine{},
+		&services.MockStateStorage{},
+		cfg,
+		log,
+		metricFactory)
+
+	t.Run("should return ok for valid block", func(t *testing.T) {
+		t.Skipf("Skipped till previous block hash code can be fixed, or maybe this whole test is wrong because it injects WithPrevBlockHash() instead of letting the code calculate it")
+		prev1Hash := digest.CalcTransactionsBlockHash(prevBlock1.TransactionsBlock)
+
+		validBlock := builders.
+			BlockPair().
```

Comment:
you can create a "preset" builder function local to this test, with all of the correct data, and then override only the "interesting" stuff per test case

Reply:
Will be part of a different PR, in branch feature/refactor-block-validation